hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -1418,4 +1417,81 @@ class TestMORDataSource extends HoodieSparkClientTestBase with SparkDatasetMixin
     metaClient = createMetaClient(spark, basePath)
     assertEquals(metaClient.getTableConfig.getRecordMergerStrategy, mergerStrategyName)
   }
+
+  /**
+   * Test Read-Optimized query on MOR table with RECORD_INDEX enabled.
+   */
+  @Test
+  def testReadOptimizedQueryWithRecordIndex(): Unit = {
```

Comment:
Rename to testQueryTypesWithRLI?

Reply:
renamed to better reflect the test purpose